### PR TITLE
Caichao/support opengles

### PR DESCRIPTION
**Pull request by CaichaoGitHub**
compatible with es2 and es3 in android system , even in complex render pipeline context  and paired with a brand new  skity demo

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -27,6 +27,19 @@ else()
   add_library(skity SHARED)
 endif()
 
+if(ANDROID)
+  find_library(
+          gles-lib
+          GLESv3
+  )
+  find_library(
```
Comment (RuiwenTang) [inline CMakeLists.txt:35]: The `proc_loader` property in GPUContext is used to load required GL function at runtime. So during compile time, skity does not need to link gles or egl library. I have a seperate android example project to show this usage, and I hope skity not link libEGL.so or libGLESv3.so at compile time.

File: example/Android/app/src/main/jni/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+
+
+cmake_minimum_required(VERSION 3.13)
+
+#add_subdirectory(../../../../../../CMakeLists.txt)
+add_subdirectory(../../../../../../ ../../../../../../../build)
+include_directories(../../../../../../include)
```
Comment (RuiwenTang) [inline example/Android/app/src/main/jni/CMakeLists.txt:7]: In project cmake `skity::skity`  defines target_include_directory with public property to specify public header fine include, and I think no need to add `include_directories` again at this posiiton.

File: CMakeLists.txt (hunk 1 above)
Comment (CaichaoGitHub) [reply CMakeLists.txt:35]: OK, i will fix it

Comment (RuiwenTang) [reply CMakeLists.txt:35]: https://github.com/RuiwenTang/Skity-Android

Comment (CaichaoGitHub) [reply CMakeLists.txt:35]: ok , i will

**Comment by CaichaoGitHub** [comment]
i would close the pr cause i thought you demo's solution was more reasonable

**Pull request closed without merging**